lisa/base_tools/sed.py

```diff
@@ -47,18 +47,29 @@ def append(
         self,
         text: str,
         file: str,
+        match_line: str = "",
         sudo: bool = False,
     ) -> None:
         # always force run, make sure it happens every time.
         text = text.replace('"', '\\"')
-        result = self.run(
-            f"-i.bak '$a{text}' {file}",
-            force_run=True,
-            no_error_log=True,
-            no_info_log=True,
-            sudo=sudo,
-            shell=True,
-        )
+        if match_line:
```

Comment:
the `if/else` should include only the different part.

```python
       if match_line:
            pattern = f"{match_line}a{text}"
        else:
            pattern = f"$a{text}"
        result = self.run(
            f"-i.bak '{pattern}' {file}",
            force_run=True,
            no_error_log=True,
            no_info_log=True,
            sudo=sudo,
            shell=True,
        )
```

Reply:
The change is done, thanks